File: baekjoon_problems/1932.py
import sys

N = int(sys.stdin.readline())

tri = []
for i in range(N):
  tri.append(list(map(int,sys.stdin.readline().split())))

tri = tri[::-1]

#print(tri)

for i in range(1,N):
  for j in range(len(tri[i])):
    tri[i][j] = max(tri[i][j] + tri[i-1][j], tri[i][j] + tri[i-1][j+1])


print(tri[-1][0])

File: baekjoon_problems_kotlin/1932.kt
import kotlin.math.max

fun main(args: Array<String>) {
    val br = System.`in`.bufferedReader()

    val height = br.readLine().toInt()
    var stairs = mutableListOf<List<Int>>()
    var stairsDp = mutableListOf<List<Int>>()

    for(i in 0 until height) {
        stairs.add(br.readLine().split(" ").map { it.toInt() })
    }

    stairs = stairs.reversed().toMutableList()
    stairsDp.add(stairs[0])

    for(i in 1 until height) {
        var temp = mutableListOf<Int>()
        for(j in 0 until stairs[i].size) {
            val bestChoice = max(stairs[i][j] + stairsDp[i-1][j], stairs[i][j] + stairsDp[i-1][j+1])
            temp.add(bestChoice)
        }
        stairsDp.add(temp)
    }

    println(stairsDp[height-1][0])



}




